File: Clients/src/protocol.cpp
#include "protocol.h"

 std::string   Protocol::ToServer(std::vector<std::string> message)
{
    std::string startChar = "&";
    std::string endChar = "#";
    std::string messageToServer = makeString(message, startChar, endChar);
    return messageToServer;
}

 std::string   Protocol::ToServer(std::string message)
{
    std::string startChar = "&";
    std::string endChar = "#";
    std::string messageToServer = makeString(message, startChar, endChar);
    return messageToServer;
}

std::vector<std::string> Protocol::FromServer(std::string message)
{
std::vector<std::string> vector = SplitString(message, ';');
return vector;
}

std::vector<std::string> Protocol::SplitString(std::string text, char split)
{
    text += " ";
    std::vector<std::string> commando;
    size_t prevIndex = 0;

    while(true)
    {
        size_t index = text.find_first_of(split, prevIndex);
        if(index > text.length())
        {
            break;
        }
        std::string ff = text.substr(prevIndex, index - prevIndex);
        commando.push_back(ff);
        prevIndex = index + 1;
    }

    return commando;
}



std::string Protocol::makeString(std::vector<std::string> message, std::string startChar, std::string endChar)
{
    std::string newString;
    newString = startChar;
    for(size_t i = 0; i < message.size(); i++)
    {
        if(i != 0)
        {
            newString += ";";
        }
        newString += message[i];

    }
    newString += endChar;
    return newString;
}

std::string Protocol::makeString(std::string message, std::string startChar, std::string endChar)
{
    std::string newString;
    newString = startChar;
    for(size_t i = 0; i < message.size(); i++)
    {
        if(i != 0)
        {
            newString += ";";
        }
        newString += message[i];

    }
    newString += endChar;
    return newString;
}

File: Control_Panel_Systeem/Control_Panel_Systeem/Protocol.cs
using System.Collections;
using System.Collections.Generic;

public class Protocol
{

    char seperatorChar = ';';
    void SplitString()
    {

    }

    string MakeString(int code, List<string> value, char startChar, char endChar)
{
    string newString;
    newString = startChar.ToString();
    newString += code;

    foreach(string mess in value)
    {
        newString += seperatorChar;
        newString += mess;
    }
    newString += endChar;
    return newString;
}

List<string> SplitString(string text, char splitChar, char startChar, char endChar)
{
    if ((text[0] == startChar) && (text[text.Length - 1] == endChar))
    {
        text = text.Substring(1, text.Length - 2);
            List<string> returnvalue = new List<string>();
            returnvalue.AddRange(text.Split(splitChar));
        return returnvalue;
    }
    List<string> fu = new List<string>();
    return fu;
}




}

public enum Codes
{
    CONNECT,CONSOLE, TOTALWATER, TOTALPOWER, GETCLIENTS, SETCLIENT, DISABLEALLCLIENTS
}